client/src/tasks/SasTaskProvider.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright © 2022, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as vscode from "vscode";
+import {
+  Execute,
+  SAS_TASK_TYPE,
+  getAllTasksFromSasTasksConfiguration,
+  runCurrentFileTask,
+  runSasFileTask,
+  runSelectedTextTask,
+} from "./SasTasks";
+
+export class SasTaskProvider implements vscode.TaskProvider {
+  provideTasks(): vscode.ProviderResult<vscode.Task[]> {
+    return getAllTasks();
+  }
+  resolveTask(): vscode.ProviderResult<vscode.Task> {
+    return undefined;
+  }
+}
+
+export class SasPseudoterminal implements vscode.Pseudoterminal {
+  constructor(
+    private execute: Execute,
+    private task: vscode.TaskDefinition,
+  ) {}
+  private messageEmitter = new vscode.EventEmitter<string>();
+  private closeEmitter = new vscode.EventEmitter<number>();
+  onDidWrite: vscode.Event<string> = this.messageEmitter.event;
+  onDidOverrideDimensions?: vscode.Event<vscode.TerminalDimensions>;
+  onDidClose?: vscode.Event<number | void> = this.closeEmitter.event;
+  onDidChangeName?: vscode.Event<string>;
+  open(): void {
+    this.executeTask();
+  }
+  close(): void {}
+  handleInput?(): void {}
+  setDimensions?(): void {}
```

Comment:
Can we update this such that we have...
 - variables first
 - constructor
 - spaces between functions